### DISCUSSION TITLE
@ledgerhq/wallet-api-client not working properly

### DISCUSSION THREAD

**Discussion by blackpanther1881** [LedgerHQ/wallet-api#244] [Q&A] [answered]
I am facing issue while calling hooks from @ledgerhq/wallet-api-client-react.  I'm just following this doc https://wallet.api.live.ledger.com/react/configuration.

package versions-  
"@ledgerhq/wallet-api-client": "^1.1.0",
"@ledgerhq/wallet-api-client-react": "^1.0.4",

this is transport
<img width="796" alt="Screenshot 2023-10-13 at 10 40 07 AM" src="https://github.com/LedgerHQ/wallet-api/assets/66773204/12f6fb5b-a2c7-4f44-bb86-da9b16c8dc8f">

wrapping app with provider
<img width="677" alt="Screenshot 2023-10-13 at 10 40 57 AM" src="https://github.com/LedgerHQ/wallet-api/assets/66773204/927c0940-4b17-406a-96eb-481c6ec90d51">

after I'm trying to access the account 
 const { accounts, loading, error } = useAccounts();
console.log(accounts, "accounts");
I'm getting accounts as null

**Comment by ComradeAERGO** [comment]
Hi @blackpanther1881 
Did you setup a manifest?

If not I apologize if the documentation is not clear about that part.
You can check the documentation [here](https://wallet.api.live.ledger.com/examples/test-live-app/manifest) to do so.

You'll need to check that your apiVersion is set to 2.0.0
And that your permissions are correctly set.

Alternatively, you can also try using the simulator.

**Reply by blackpanther1881** [reply]
Hello @ComradeAERGO updated manifest apiVersion to 2.0.0

I'm directly using ledger live not using simulator. I'm getting accounts list as null even after updating apiVersion to 2.0.0

My manifest.json
```
{
      "id": "pstake-cosmos-dapp",
      "name": "pSTAKE-COSMOS",
      "url": "https://dapp-browser.apps.ledger.com/v2",
      "params": {
        "dappUrl": "http://localhost:3000",
        "nanoApp": "pSTAKE",
        "dappName": "pSTAKE",
        "networks": [
          {
            "currency": "cosmos",
            "chainID": "cosmoshub-4",
            "nodeURL": "https://rpc.cosmos.audit.one"
          },
          {
            "currency": "xprt",
            "chainID": "core-1",
            "nodeURL": "https://rpc.core.persistence.one"
          }
        ]
      },
      "homepageUrl": "https://pstake.finance/",
      "icon": "https://cosmos.pstake.finance/images/tokens/stk_atom.svg",
      "platform": ["ios","android","desktop"],
      "apiVersion": "2.0.0",
      "manifestVersion": "1",
      "branch": "stable",
      "categories": ["staking", "defi"],
      "currencies": "*",
      "content": {
        "shortDescription": {
          "en": "Liquid Staking for The Internet of Blockchains"
        },
        "description": {
          "en": "Liquid Staking for The Internet of Blockchains"
        }
      },
      "permissions": [
        "account.list",
        "account.receive",
        "account.request",
        "currency.list",
        "device.close",
        "device.exchange",
        "device.transport",
        "message.sign",
        "transaction.sign",
        "transaction.signAndBroadcast",
        "wallet.capabilities",
        "wallet.info",
        "wallet.userId"
      ],
      "domains": ["https://*"],
      "visibility": "complete"
    }
```

**Reply by ComradeAERGO** [reply, marked as answer]
Thanks for sharing your manifest.

From what I see, you are pointing to the eth-dapp-browser in your url field,
you should be pointing to your app's url instead.

The Dapp Browser is another embedding method in Ledger Live that isn't based on the wallet-api's integration and reserved to Ethereum.

I apologies for the confusion, as we are currently reworking the documentation on that matter.

Please replace
`"url": "https://dapp-browser.apps.ledger.com/v2",`
with
`"url": "https://localhost:3000",`

**Reply by blackpanther1881** [reply]
Thanks a lot it's giving accounts now. pls modify docs according to the network. it's really hard to configure with other networks like cosmos.
One thing not clear, do you have any example cosmos transaction code. All the example transactions mentioned in docs was ethereum network txns.

**Reply by ComradeAERGO** [reply]
You're welcome. I've made sure the update on the doc tackles the ambiguity regarding de dappBrowser.

As for cosmos transaction code, I'll have to ask, I'll come back to you asap.

**Reply by blackpanther1881** [reply]
@ComradeAERGO any update on this

**Reply by agusseptiono1992** [reply]
74647a4fc01f7a50c0ed18ad4e341bb56c1c12aa

**Reply by harry32459** [reply]
Best way to test Wallet API error handling under network stress?

**Reply by oliviauk842-bit** [reply]
I use Toxiproxy to simulate flaky networks. Great for timeout testing

**Reply by JudeStorm** [reply]
@harry32459 Same here. www.comparenodes.com benchmarks show how Wallet-APIs behave when latency spikes

**Reply by adriana-steele** [reply]
I log retries + fallbacks, then match with https://comparenodes.com logs. Found silent request drops under load .
